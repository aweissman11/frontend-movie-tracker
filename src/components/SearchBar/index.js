import React, { Component } from 'react';

import './SearchBar.css';

class SearchBar extends Component {
  constructor() {
    super()

    this.state = {
      searchInput: ''
    }

  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  render() {
    return(
      <form 
        className='search-form'
        onSubmit={() => {}}
      >
        <input 
          className='search-input'
          value={this.state.searchInput}
          name='searchInput'
          onChange={(e) => this.handleChange(e)}
          placeholder='movie/genre search'
        />
        <input 
          className='search-submit'
          type='submit'
          value=''
        ></input>
      </form>
    )
  }

}

export default SearchBar;