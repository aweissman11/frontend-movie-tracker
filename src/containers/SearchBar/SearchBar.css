.search-form {
  height: 36px;
  box-sizing: border-box;
}

.search-form .search-submit {
  background: url(./images/search.png);
  background-size: contain;
  background-repeat: no-repeat;
  padding-top: 10px;
  width: 25px;
  height: 25px;
  border: none;
  padding: 5px 0;
  transition: .3s;
}

.search-form .search-submit:hover {
  transform: rotate(90deg);
  cursor: pointer;
}

.search-form .search-input {
  border: 2px solid #d1d1d1;
  width: 250px;
  margin: 0 .5rem;
  padding-left: 4px;
  height: 30px;
  font-size: 14px;
  color: #d1d1d1;
}

.search-form .search-input:placeholder {
  color: #d1d1d1;
}