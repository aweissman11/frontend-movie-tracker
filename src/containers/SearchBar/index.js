import React, { Component } from 'react';
import { connect } from 'react-redux';

import * as Cleaners from '../../utilities/cleaners';
import fetchCall from '../../utilities/fetchCall';
// import { genres, ratings, sortOptions } from './filtersInfo';
import { getMovieList } from '../../actions/thunkActions/SearchBarThunk';


import './SearchBar.css';

class SearchBar extends Component {
  constructor() {
    super()

    this.state = {
      searchInput: ''
    }

  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleSubmitSearch = async (e) => {
    e.preventDefault();
    this.props.setFetchedMovies(this.state.searchInput)
  }

  render() {
    return(
      <form 
        className='search-form'
        onSubmit={this.handleSubmitSearch}
      >
        <input 
          className='search-input'
          value={this.state.searchInput}
          name='searchInput'
          onChange={(e) => this.handleChange(e)}
          placeholder='movie/genre search'
        />
        <input 
          className='search-submit'
          type='submit'
          value=''
        ></input>
      </form>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setFetchedMovies: (data) => dispatch(getMovieList(data)),
});

export default connect(null, mapDispatchToProps)(SearchBar);