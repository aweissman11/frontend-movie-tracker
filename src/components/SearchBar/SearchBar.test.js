import React from 'react'
import SearchBar from './';

import { shallow } from 'enzyme';

describe('SearchBar', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SearchBar />);
  });

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state()).toEqual({searchInput: ''});
  });

  it('should call handleChange on text input', () => {
    const event = {target: {
      value: 'a', 
      name: 'searchInput'
    }};
    const mockHandleChange = jest.fn();
    wrapper.instance().handleChange = mockHandleChange;

    wrapper.find('.search-input').simulate('change', event);

    expect(mockHandleChange).toHaveBeenCalledWith(event);
  });

  it('should set state when handleChange is called', () => {
    const event = {target: {
      value: 'a', 
      name: 'searchInput'
    }};

    wrapper.instance().handleChange(event);

    expect(wrapper.state().searchInput).toEqual('a');
  });

});