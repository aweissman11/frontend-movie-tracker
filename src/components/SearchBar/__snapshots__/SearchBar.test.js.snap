// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match the snapShot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="search-input"
          name="searchInput"
          onChange={[Function]}
          placeholder="movie/genre search"
          value=""
        />,
        <input
          className="search-submit"
          type="submit"
          value=""
        />,
      ],
      "className": "search-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-input",
          "name": "searchInput",
          "onChange": [Function],
          "placeholder": "movie/genre search",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search-submit",
          "type": "submit",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="search-input"
            name="searchInput"
            onChange={[Function]}
            placeholder="movie/genre search"
            value=""
          />,
          <input
            className="search-submit"
            type="submit"
            value=""
          />,
        ],
        "className": "search-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-input",
            "name": "searchInput",
            "onChange": [Function],
            "placeholder": "movie/genre search",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search-submit",
            "type": "submit",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
